platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,493 @@
+     * This list SHALL NOT be empty.<br/>
+     * This list SHALL NOT contain more than `10` entries.<br/>
+     * The first two entries in this list SHALL be the endpoints published to
+     * all consensus nodes.<br/>
+     * All other entries SHALL be reserved for future use.
+     */
+    repeated proto.ServiceEndpoint gossip_endpoint = 5;
+}
+```
+
+#### Roster Map
+
+A States API map `Roster Map` (separate from the PBJ Map modeled above) will be introduced.
+It will contain keys which will be the hash of the roster (of type `byte`) and values of type `Roster`.
+This map will store all candidate and active rosters.
+It will be an implementation of `ReadableKVState<>` and `WriteableKVState<>` interfaces.
+
+The `byte` key of the Roster Map will be of the widely used type `com.hedera.pbj.runtime.io.buffer.Bytes`.
+
+#### Roster State
+
+The `RosterState` protobuf is defined as follows.
+
+```proto
+/**
+ * The current state of platform rosters.<br/>
+ * This message stores a roster data for the platform in network state.
+ *
+ * The roster state SHALL encapsulate the incoming candidate roster's hash,<br/>
+ * and a list of pairs of round number and active roster hash.<br/>
+ * This data SHALL be used to track round numbers and the rosters used in determining the consensus.<br/>
+ */
+message RosterState {
+
+    /**
+     * The SHA-384 hash of a candidate roster.
+     * <p>
+     * This is the hash of the roster that is currently being considered
+     * for adoption.
+     * A Node SHALL NOT, ever, have more than one candidate roster
+     * at the same time.
+     */
+    byte candidate_roster_hash = 1;
+
+    /**
+     * The SHA-384 hash of an active roster
+     * This is the hash of the roster that has already been adopted by the
+     * network.
+     * A Node SHALL NOT, ever, have more than one active roster
+     * at the same time.
+     */
+    repeated RoundRosterPair round_id_to_roster_pair = 2;
+}
+```
+
+#### RoundRosterPair
+
+A `RoundRosterPair` will be defined as follows:
+
+```proto
+/**
+ * A pair of round number and active roster hash.
+ * <p>
+ * This message SHALL encapsulate the round number and the hash of the
+ * active roster used for that round.
+ */
+message RoundRosterPair {
+
+    /**
+     * The round number.
+     * <p>
+     * This value SHALL be the round number of the consensus round.
+     */
+    uint64 round_number = 1;
+
+    /**
+     * The SHA-384 hash of the active roster for the given round number.
+     * <p>
+     * This value SHALL be the hash of the active roster used for the round.
+     */
+    byte active_roster_hash = 2;
+}
+```
+It is noteworthy that the roster must be immutable to guarantee the integrity of the computed hash.
+This map of rosters will typically contain the current Active Roster, an optional previously Active Roster, and an
+optional current Candidate Roster. Insertion of rosters will be controlled by ensuring that the acceptance of a new
```

Comment:
Doesn't the candidate roster goes in the RosterMap, not the RosterState? If i'm wrong, then the comment for `active_roster_hash` and the field name must change.